### PR TITLE
Read oauth2 secret from file

### PR DESCRIPTION
**Pull request by roidelapluie**
cc @LeviHarrison

I still need to add tests.

Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

**Comment by roidelapluie** [comment]
I also need to add a check that a user does not pass the client_secret and the client_secret_file

File: config/http_config.go [hunk 1]
```diff
@@ -462,20 +463,70 @@ func (rt *basicAuthRoundTripper) CloseIdleConnections() {
...
+		changed bool
+	)
+
+	if rt.config.ClientSecretFile != "" {
+		data, err := ioutil.ReadFile(rt.config.ClientSecretFile)
+		if err != nil {
+			return nil, fmt.Errorf("unable to read oauth2 client secret file %s: %s", rt.config.ClientSecretFile, err)
+		}
+		secret = string(data)
+		changed = secret != rt.secret
```
Comment (roidelapluie) [inline config/http_config.go:493]: todo: add rlock here

**Comment by roidelapluie** [comment]
Todo: strip extra blank chars

**Comment by LeviHarrison** [comment]
Everything so far looks good!

File: config/http_config.go [hunk 2]
```diff
@@ -489,8 +494,10 @@ func (rt *oauth2RoundTripper) RoundTrip(req *http.Request) (*http.Response, erro
 		if err != nil {
```
Comment (LeviHarrison) [inline config/http_config.go:494]: Perhaps should we lock over here instead so we don't get multiple threads reading the file at once and assessing the same thing?

**Review by LeviHarrison — COMMENTED** [review]
One more thing that GitHub won't let me add a comment to. Do we need to add OAuth2 here now that it can read from file?

https://github.com/roidelapluie/common/blob/9361f0ddb7fc9336498d0569e635add8153cf786/config/http_config.go#L148-L156

File: config/http_config.go (hunk 2 above)
Comment (LeviHarrison) [reply config/http_config.go:494]: Just above that line, once again GitHub not letting me comment

Comment (LeviHarrison) [reply config/http_config.go:494]: Actually wait never mind that's pointless.

**Comment by roidelapluie** [comment]
Yes thanks!!!

**Review by LeviHarrison — APPROVED** [review]
Looks great!

**Pull request merged**